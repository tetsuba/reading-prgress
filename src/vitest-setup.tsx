import { QueryClient, QueryClientProvider } from 'react-query'
import { MemoryRouter } from 'react-router-dom'
import { Provider } from 'react-redux'
import store from './store/store'
import { updateUser } from './store/user/userSlice'
import { addBooks } from './store/books/booksSlice'
import { addStudents } from './store/students/studentsSlice'
import { studentsMockData } from '../tests/mockData/students'
import booksMockData from '../tests/mockData/books'
import { userMockData } from '../tests/mockData/user'

const portal = document.createElement('div')
portal.setAttribute('id', 'modal')
document.body.appendChild(portal)

const mockResults = {
    results: [[{ transcript: 'This is a story' }]]
}

class MockSpeech {
    start() {
        // @ts-ignore
        this.onresult(mockResults)
    }
    stop = vi.fn()
    onresult = vi.fn()
    onend = vi.fn()
}

Object.defineProperty(window, 'webkitSpeechRecognition', {
    writable: true,
    value: vi.fn().mockImplementation(() => new MockSpeech())
})

const queryClient = new QueryClient({
    defaultOptions: {
        queries: {
            staleTime: 0,
            cacheTime: 0,
            retryDelay: 1,
            retry: 0
        }
    }
})

type PropTypes = {
    pathname: string
    children: JSX.Element
    notAuthenticated?: boolean
}

export function updateStoreWithMockData() {
    store.dispatch(updateUser(userMockData))
    store.dispatch(addBooks(booksMockData))
    store.dispatch(addStudents(studentsMockData))
}

export function WrapperWith_Store_Query_Router(props: PropTypes) {
    if (!props.notAuthenticated) updateStoreWithMockData()
    return (
        <Provider store={store}>
            <QueryClientProvider client={queryClient}>
                <MemoryRouter initialEntries={[props.pathname]}>
                    {props.children}
                </MemoryRouter>
            </QueryClientProvider>
        </Provider>
    )
}

export function WrapperWith_Store_Router(props: PropTypes) {
    return (
        <Provider store={store}>
            <MemoryRouter initialEntries={[props.pathname]}>
                {props.children}
            </MemoryRouter>
        </Provider>
    )
}

export function WrapperWithQueryAndRouter(props: PropTypes) {
    return (
        <QueryClientProvider client={queryClient}>
            <MemoryRouter initialEntries={[props.pathname]}>
                {props.children}
            </MemoryRouter>
        </QueryClientProvider>
    )
}

export function WrapperWithQuery(props: { children: JSX.Element }) {
    return (
        <QueryClientProvider client={queryClient}>
            {props.children}
        </QueryClientProvider>
    )
}

export function WrapperWithRouter(props: PropTypes) {
    return (
        <MemoryRouter initialEntries={[props.pathname]}>
            {props.children}
        </MemoryRouter>
    )
}
