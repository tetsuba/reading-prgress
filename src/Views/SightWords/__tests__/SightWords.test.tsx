import { render, screen, waitFor } from '@testing-library/react'
import { WrapperWith_Store_Query_Router } from '../../../vitest-setup'
import { Mocked } from 'vitest'

import SiteWords from '../SightWords'

import axios from 'axios'
vi.mock('axios')
const mockedAxios = axios as Mocked<typeof axios>

const mockData = {
    data: {
        sightWordsReadWrong: [
            { word: 'this', index: 1 },
            { word: 'that', index: 5 }
        ],
        sightWordsReadInBooks: [
            { word: 'this', index: 1 },
            { word: 'that', index: 5 },
            { word: 'would', index: 10 }
        ],
        sightWordsFromBooks: [
            { word: 'this', index: 1 },
            { word: 'that', index: 5 },
            { word: 'would', index: 10 },
            { word: 'dog', index: 15 }
        ],
        sightWordsNotInBooks: [
            { word: 'right', index: 20 },
            { word: 'live', index: 40 }
        ]
    }
}

describe('SightWords', () => {
    test('will render groups of sight words', async () => {
        mockedAxios.get.mockResolvedValue(mockData)
        const { asFragment } = render(
            <WrapperWith_Store_Query_Router pathname={'/site-words'}>
                <SiteWords />
            </WrapperWith_Store_Query_Router>
        )
        await waitFor(() =>
            expect(screen.queryAllByTestId('heat-map')).toHaveLength(4)
        )
        expect(asFragment()).toMatchSnapshot()
    })
})
