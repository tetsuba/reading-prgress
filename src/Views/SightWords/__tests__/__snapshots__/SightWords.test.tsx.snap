// Vitest Snapshot v1

exports[`SightWords > will render groups of sight words 1`] = `
<DocumentFragment>
  <header
    class="bg-white shadow"
  >
    <div
      class="mx-auto flex max-w-7xl justify-between py-6 px-4 sm:px-6 lg:px-8"
    >
      <h1
        class="text-3xl font-bold tracking-tight text-gray-900"
      >
        Sight Words
      </h1>
    </div>
  </header>
  <main>
    <div
      class="mx-auto max-w-7xl py-6 sm:px-6 lg:px-8"
    >
      <div
        class="px-4 md:px-0"
      >
        <input
          class="bg-gray-50 border border-gray-300  text-gray-900 text-sm rounded-lg block w-full p-2.5 focus:ring-blue-500 focus:border-blue-500 "
          data-testid="search"
          defaultvalue=""
          placeholder="Search"
          template="text"
          type="text"
        />
        <h3
          class="text-ml font-medium text-gray-900 dark:text-white md:text-xl mt-10 rounded-t-lg bg-gray-200 p-2"
        >
          Sight Words Read Incorrectly
        </h3>
        <div
          class="min-h-96 md:border-3 inline-flex w-full flex-wrap items-center justify-center rounded-b-lg border-2 border-t-0 border-dashed border-gray-200 p-2 shadow-md"
          data-testid="heat-map"
        >
          <span
            class="m-2 rounded-2xl border-2 p-3 md:m-5"
            data-testid="heat-map-word"
            style="background-color: rgb(255, 204, 204);"
          >
            <span
              class="font-bold text-gray-900"
            >
              this
            </span>
          </span>
          <span
            class="m-2 rounded-2xl border-2 p-3 md:m-5"
            data-testid="heat-map-word"
            style="background-color: rgb(255, 0, 0);"
          >
            <span
              class="font-bold text-white"
            >
              that
            </span>
            <span
              class="ml-1 rounded-full py-1 px-2 text-white md:ml-3"
              style="background-color: rgb(0, 0, 0);"
            >
              5
            </span>
          </span>
        </div>
        <h3
          class="text-ml font-medium text-gray-900 dark:text-white md:text-xl mt-10 rounded-t-lg bg-gray-200 p-2"
        >
          Sight Words Read In Books
        </h3>
        <div
          class="min-h-96 md:border-3 inline-flex w-full flex-wrap items-center justify-center rounded-b-lg border-2 border-t-0 border-dashed border-gray-200 p-2 shadow-md"
          data-testid="heat-map"
        >
          <span
            class="m-2 rounded-2xl border-2 p-3 md:m-5"
            data-testid="heat-map-word"
            style="background-color: rgb(229.5, 242.25, 255);"
          >
            <span
              class="font-bold text-gray-900"
            >
              this
            </span>
          </span>
          <span
            class="m-2 rounded-2xl border-2 p-3 md:m-5"
            data-testid="heat-map-word"
            style="background-color: rgb(127.5, 191.25, 255);"
          >
            <span
              class="font-bold text-gray-900"
            >
              that
            </span>
            <span
              class="ml-1 rounded-full py-1 px-2 text-white md:ml-3"
              style="background-color: rgb(127.5, 127.5, 127.5);"
            >
              5
            </span>
          </span>
          <span
            class="m-2 rounded-2xl border-2 p-3 md:m-5"
            data-testid="heat-map-word"
            style="background-color: rgb(0, 127.5, 255);"
          >
            <span
              class="font-bold text-white"
            >
              would
            </span>
            <span
              class="ml-1 rounded-full py-1 px-2 text-white md:ml-3"
              style="background-color: rgb(0, 0, 0);"
            >
              10
            </span>
          </span>
        </div>
        <h3
          class="text-ml font-medium text-gray-900 dark:text-white md:text-xl mt-10 rounded-t-lg bg-gray-200 p-2"
        >
          Sight Words In Books
        </h3>
        <div
          class="min-h-96 md:border-3 inline-flex w-full flex-wrap items-center justify-center rounded-b-lg border-2 border-t-0 border-dashed border-gray-200 p-2 shadow-md"
          data-testid="heat-map"
        >
          <span
            class="m-2 rounded-2xl border-2 p-3 md:m-5"
            data-testid="heat-map-word"
            style="background-color: rgb(238, 249.33333333333334, 238);"
          >
            <span
              class="font-bold text-gray-900"
            >
              this
            </span>
          </span>
          <span
            class="m-2 rounded-2xl border-2 p-3 md:m-5"
            data-testid="heat-map-word"
            style="background-color: rgb(170, 226.66666666666666, 170);"
          >
            <span
              class="font-bold text-gray-900"
            >
              that
            </span>
            <span
              class="ml-1 rounded-full py-1 px-2 text-white md:ml-3"
              style="background-color: rgb(170, 170, 170);"
            >
              5
            </span>
          </span>
          <span
            class="m-2 rounded-2xl border-2 p-3 md:m-5"
            data-testid="heat-map-word"
            style="background-color: rgb(85, 198.33333333333334, 85);"
          >
            <span
              class="font-bold text-white"
            >
              would
            </span>
            <span
              class="ml-1 rounded-full py-1 px-2 text-white md:ml-3"
              style="background-color: rgb(85, 85, 85);"
            >
              10
            </span>
          </span>
          <span
            class="m-2 rounded-2xl border-2 p-3 md:m-5"
            data-testid="heat-map-word"
            style="background-color: rgb(0, 170, 0);"
          >
            <span
              class="font-bold text-white"
            >
              dog
            </span>
            <span
              class="ml-1 rounded-full py-1 px-2 text-white md:ml-3"
              style="background-color: rgb(0, 0, 0);"
            >
              15
            </span>
          </span>
        </div>
        <h3
          class="text-ml font-medium text-gray-900 dark:text-white md:text-xl mt-10 rounded-t-lg bg-gray-200 p-2"
        >
          Sight Words Not Available In Books
        </h3>
        <div
          class="min-h-96 md:border-3 inline-flex w-full flex-wrap items-center justify-center rounded-b-lg border-2 border-t-0 border-dashed border-gray-200 p-2 shadow-md"
          data-testid="heat-map"
        >
          <span
            class="m-2 rounded-2xl border-2 p-3 md:m-5"
            data-testid="heat-map-word"
            style="background-color: rgb(230, 230, 230);"
          >
            <span
              class="font-bold text-gray-900"
            >
              right
            </span>
            <span
              class="ml-1 rounded-full py-1 px-2 text-white md:ml-3"
              style="background-color: rgb(127.5, 127.5, 127.5);"
            >
              20
            </span>
          </span>
          <span
            class="m-2 rounded-2xl border-2 p-3 md:m-5"
            data-testid="heat-map-word"
            style="background-color: rgb(230, 230, 230);"
          >
            <span
              class="font-bold text-gray-900"
            >
              live
            </span>
            <span
              class="ml-1 rounded-full py-1 px-2 text-white md:ml-3"
              style="background-color: rgb(0, 0, 0);"
            >
              40
            </span>
          </span>
        </div>
      </div>
    </div>
  </main>
</DocumentFragment>
`;
